### PR TITLE
Added passwords, refactored the join game menu.

### PR DESCRIPTION
**Pull request by admoore0**

### CODE REVIEW

**Review by kdreese — DISMISSED** [review]

File: src/states/menus/create_game_menu.gd [hunk 1]
```diff
@@ -41,8 +43,16 @@ func _ready() -> void:
 	port_spin_box.value = Global.config.get("port", 0)
 
 
+func reset_create_button() -> void:
+	if mode == CreateMode.HOST_LOCAL:
+		create_button.disabled = false
+	else:
+		create_button.disabled = (name_line_edit.text == "")
```
Comment (thearst3rd) [inline src/states/menus/create_game_menu.gd:50]: Nerd ass nitpick 🤓🤓
```suggestion
		create_button.disabled = name_line_edit.text == ""
```

File: src/states/menus/join_game_menu.gd [hunk 2]
```diff
@@ -42,11 +44,20 @@ func _ready() -> void:
...
+	row = server_grid.get_child(0)
 	server_grid.remove_child(row)
+	# Remove the second row.
+	server_grid.remove_child(server_grid.get_child(0))
 	button_group.pressed.connect(on_radio_button_pressed)
 
 
+func reset_join_button() -> void:
+	if mode == JoinMode.JOIN_LOCAL:
+		join_button.disabled = (host_line_edit.text == "")
```
Comment (thearst3rd) [inline src/states/menus/join_game_menu.gd:56]: 🤓
```suggestion
		join_button.disabled = host_line_edit.text == ""
```

**Review by thearst3rd — DISMISSED** [review]
freaking nerd emojis

File: src/autoload/gmp_client.gd [hunk 3]
```diff
@@ -37,17 +38,20 @@ func make_request(host: String, method: HTTPClient.Method, request: Dictionary)
 	remove_child(http_request)
 
 	if http_response[0]:
+		push_error("Error receiving HTTP response.")
 		return[http_response[0], {"error": "Could not connect to server."}]
 
-	# HTTP code 500 and above indicate that the server is probably down.
-	if http_response[1] >= 500:
+	# HTTP code 502 and 504 and above indicate that the server is probably down.
```
Comment (kdreese) [inline src/autoload/gmp_client.gd]: Extreme pedantry

**Review by kdreese — DISMISSED** [review]

**Review by kdreese — APPROVED** [review]

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**